hnn_core/network.py

```diff
@@ -251,8 +251,58 @@ def _add_external_feeds(self):
         unique_keys = sorted(self.p_unique.keys())
         self.feedname_list += unique_keys
 
-    def _create_gid_dict(self):
-        """Creates gid dicts."""
+        self._update_gid_dict()  # add feeds -> update the gid_dict
+        self.instantiate_feeds()
+
+    def instantiate_feeds(self):
+        # each trial needs unique event time vectors
+
+        self._tmp_feed_event_times = []  # reset if called again
+        n_trials = self.params['N_trials']
+
+        cur_params = self.params.copy()  # these get mangled below!
+        for trial_idx in range(n_trials):
+
+            prng_seedcore_initial = cur_params['prng_*'].copy()
+            for param_key in prng_seedcore_initial.keys():
+                cur_params[param_key] =\
+                    prng_seedcore_initial[param_key] + trial_idx
+            # needs to be re-run to create the dicts going into ExtFeed
+            p_common, p_unique = create_pext(cur_params,
+                                             cur_params['tstop'])
+
+            src_types = self.feedname_list
+            event_times_per_source = {}
+            for src_type in src_types:
+                event_times = []
+                if src_type == 'common':
+                    for idx, gid in enumerate(self.gid_dict['common']):
+                        gid_target = None  # 'common' attaches to all
+                        feed = ExtFeed(feed_type=src_type,
+                                       target_cell_type=gid_target,
+                                       params=p_common[idx],
+                                       gid=gid)
+                        event_times.append(feed.event_times)
+
+                elif src_type in p_unique.keys():
+                    for gid in self.gid_dict[src_type]:
+                        gid_target = gid - self.gid_dict[src_type][0]
+                        target_cell_type = self.gid_to_type(gid_target)
+                        feed = ExtFeed(feed_type=src_type,
+                                       target_cell_type=target_cell_type,
+                                       params=p_unique[src_type],
+                                       gid=gid)
+                        event_times.append(feed.event_times)
+                event_times_per_source.update({src_type: event_times})
+            self._tmp_feed_event_times.append(event_times_per_source.copy())
+
+    def _update_gid_dict(self):
```

Comment:
pass arguments explicitly?

Reply:
-1 The reliance on `pos_dict` for all inference is weird and should be fixed. I think it's more than just changing the calling pattern of this method.